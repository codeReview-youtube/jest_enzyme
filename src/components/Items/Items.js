import React from 'react';
import Item from '../Item/Item';
import './items.scss';

/**
 * @author
 * @function Items
 **/

const Items = ({ items, onClick, onDelete, onEdit }) => {
  return (
    <div className='list'>
      {items.map((item, index) => (
        <Item
          item={item}
          key={`${item.id}-${index}`}
          onClick={() => onClick(index)}
          onDelete={() => onDelete(index)}
          onEdit={() => onEdit(index)}
        />
      ))}
    </div>
  );
};

export default Items;
