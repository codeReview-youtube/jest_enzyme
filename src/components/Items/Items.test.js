/**
 * @jest-environment jsdom
 */
/* eslint-disable react/display-name */
import React from 'react';
import { shallow } from 'enzyme';
/**
 *import  Component
 */
import Items from './Items';
/**
 * props
 */
const props = {
  items: [
    {
      id: 1,
      title: 'first Todo',
      completed: false
    }
  ],
  onDelete: jest.fn(),
  onEdit: jest.fn(),
  onClick: jest.fn()
};
/**
 * Tests: Items Component
 */
describe('Items Component', () => {
  /**
   * Snapshot:[default]
   */
  test('should match default snapshot', () => {
    expect.assertions(1);

    const wrapper = shallow(<Items {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
  /**
   * triggers [onClick]
   */
  test('should trigger onClick', () => {
    expect.assertions(1);
    const index = 0;

    const wrapper = shallow(<Items {...props} />);
    wrapper
      .find('Item')
      .at(index)
      .prop('onClick')();

    expect(props.onClick).toHaveBeenCalledWith(index);
  });
});
