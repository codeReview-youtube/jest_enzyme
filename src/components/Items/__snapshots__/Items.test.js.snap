// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items Component should match default snapshot 1`] = `
<div
  className="list"
>
  <Item
    item={
      Object {
        "completed": false,
        "id": 1,
        "title": "first Todo",
      }
    }
    key="1-0"
    onClick={[Function]}
    onDelete={[Function]}
    onEdit={[Function]}
  />
</div>
`;
